### PR TITLE
MTA-3334 -JAVA annotation

### PR DESCRIPTION
**Pull request by vsodhi**
JIRA: [MTA-3334](https://issues.redhat.com/browse/MTA-3334)
Initial google doc draft: https://docs.google.com/document/d/1pYq-3yVtaPSsD4Ybzuit_jqaN47JghKUSSoKQoGVops/edit

### CODE REVIEW

**Comment by jmle** [comment]
LGTM :+1:

**Review by sshveta — APPROVED** [review]

File: docs/topics/yaml-rule-structure-syntax.adoc [hunk 1]
```diff
@@ -361,17 +361,28 @@ This provider has the following capabilities:
 
 .`referenced`
 
-The `referenced` capability enables the provider to find references in the source code. This capability takes two input parameters, `pattern` and `location`.
+The `referenced` capability enables the provider to find references in the source code. This capability takes three input parameters, `pattern`, `location`, and 'annotated'.
```
Comment (anarnold97) [inline docs/topics/yaml-rule-structure-syntax.adoc:364]:
```suggestion
The `referenced` capability enables the provider to find references in the source code. This capability takes three input parameters: `pattern`, `location`, and 'annotated'.
```

File: docs/topics/yaml-rule-structure-syntax.adoc [hunk 2]
```diff
@@ -361,17 +361,28 @@ This provider has the following capabilities:
...
 ----
 when:
   java.referenced:
     pattern: "<pattern>" <1>
     location: "<location>" <2>
+    annotated: "<annotated>" <3>
 ----
 <1> A regular expression pattern to match, for example, `org.kubernetes.*`.
 <2> Specifies the exact location where the pattern needs to be matched, for example, `IMPORT`.
+<3> Checks for specific annotations and their elements, such as name and value in the Java code using a query. For example, the following query matches the Bean(url = “http://www.example.com”) annotation in the method.
```
Comment (anarnold97) [inline docs/topics/yaml-rule-structure-syntax.adoc:376]:
```suggestion
<3> Checks for specific annotations and their elements, such as name and value in the Java code using a query. For example, the following query matches the Bean (url = “http://www.example.com”) annotation in the method.
```

**Review by anarnold97 — APPROVED** [review]

File: docs/topics/yaml-rule-structure-syntax.adoc (hunk 1 above)
Comment (vsodhi) [reply docs/topics/yaml-rule-structure-syntax.adoc:364]: Good catch! Fixed.

File: docs/topics/yaml-rule-structure-syntax.adoc (hunk 2 above)
Comment (vsodhi) [reply docs/topics/yaml-rule-structure-syntax.adoc:376]: This one appears without a space in the code, so I guess it should be fine.

**Review by rromannissen — APPROVED** [review]
LGTM

**Pull request merged**